pkg/render/render.go

```diff
@@ -39,8 +39,9 @@ var (
 )
 
 type Component interface {
-	// Objects returns all objects this component contains.
-	Objects() []runtime.Object
+	// Objects returns the lists of objects in this component that should be created and/or deleted during
+	// rendering.
+	Objects() ([]runtime.Object, []runtime.Object)
```

Comment:
Maybe we can name the returned params for clarity. Ex:
`Objects() (objsToCreate, objsToDelete []runtime.Object)`